### PR TITLE
[DUOS-1958][risk=no] NIH And AnVIL Use Section

### PR DESCRIPTION
**Pull request by connorlbark**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-1958

The `FormData` components had matured quite a bit, so I recreated Shae's changes in a new branch to make the work simpler.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/data_submission/NihAnvilUse.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        {text: NO, name: 'no'},
+      ],
+      validators: [FormValidators.REQUIRED],
+      onChange: (config) => {
+
+        const value = nihAnvilUseLabels[config.value];
+        onChange({key: config.key, value: value, isValid: config.isValid});
+
+        // if going from did -> i will / no, then clear all values
+        if (nihAnvilUse == I_DID && value != I_DID) {
```
Comment (rushtong) [inline src/components/data_submission/NihAnvilUse.js]:
```suggestion
        if (nihAnvilUse === I_DID && value !== I_DID) {
```

File: src/components/data_submission/NihAnvilUse.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+const I_DID = 'I Did';
+const I_WILL = 'I Will';
+const NO = 'No';
+
+const nihAnvilUseLabels = {
+  i_did: I_DID,
+  i_will: I_WILL,
+  no: NO
+};
```
Comment (rushtong) [inline src/components/data_submission/NihAnvilUse.js:13]: One thing that I think would be helpful is to extract all of these kinds of constants to a utility function that describes ALL of the things in a dataset registration. Later on, when the UI has access to the [json schema in consent](https://github.com/DataBiosphere/consent/pull/1764), then that utility function can be updated more easily to take advantage of existing code and reduce the duplication.

**Review by rushtong — APPROVED** [review]
One syntax fix, then 👍🏽  Optionally, look into refactoring the constants for later optimization. Not required for this PR.

**Review by quazi-h — APPROVED** [review]
Changes and the new NIH and AnVIL use section on the Dataset Registration page looks good to me.

**Pull request merged**